atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/expectations/KeyValueServiceDataTrackerTest.java

```diff
@@ -0,0 +1,295 @@
+
+        assertThat(tracker.getReadInfo())
+                .isEqualTo(createTransactionReadInfo(
+                        2 * SMALL_BYTES_READ + MEDIUM_BYTES_READ + 2 * HIGH_LARGE_BYTES, totalNumberOfKvsCalls));
+
+        assertThat(tracker.getReadInfoByTable())
+                .containsExactlyEntriesOf(ImmutableMap.of(
+                        TABLE_1,
+                                createTransactionReadInfo(
+                                        SMALL_BYTES_READ + MEDIUM_BYTES_READ, numberOfKvsCallsForTable1),
+                        TABLE_2, createTransactionReadInfo(0, numberOfKvsCallsForTable2),
+                        TABLE_3,
+                                createTransactionReadInfo(
+                                        SMALL_BYTES_READ + 2 * HIGH_LARGE_BYTES, numberOfKvsCallsForTable3)));
+    }
+
+    @Test
+    public void recordTableAgnosticReadCallsAreTracked() {
+        tracker.recordTableAgnosticRead(KVS_METHOD_NAME_4, SMALL_BYTES_READ);
+        tracker.recordTableAgnosticRead(KVS_METHOD_NAME_4, HIGH_LARGE_BYTES);
+
+        tracker.recordTableAgnosticRead(KVS_METHOD_NAME_5, SMALL_BYTES_READ);
+        tracker.recordTableAgnosticRead(KVS_METHOD_NAME_5, MEDIUM_BYTES_READ);
+        tracker.recordTableAgnosticRead(KVS_METHOD_NAME_5, HIGH_LARGE_BYTES);
+
+        assertThat(tracker.getReadInfo())
+                .isEqualTo(createTransactionReadInfo(
+                        2 * SMALL_BYTES_READ + MEDIUM_BYTES_READ + 2 * HIGH_LARGE_BYTES,
+                        5,
+                        ImmutableKvsCallReadInfo.of(KVS_METHOD_NAME_4, HIGH_LARGE_BYTES)));
+
+        assertThat(tracker.getReadInfoByTable()).isEmpty();
+    }
+
+    @Test
+    public void differentMethodCallsAreTracked() {
+        tracker.recordReadForTable(TABLE_1, KVS_METHOD_NAME_1, SMALL_BYTES_READ);
+        BytesReadTracker bytesReadTracker = tracker.recordCallForTable(TABLE_1);
+        bytesReadTracker.record(MEDIUM_BYTES_READ);
+
+        tracker.recordTableAgnosticRead(KVS_METHOD_NAME_5, HIGH_LARGE_BYTES);
+
+        assertThat(tracker.getReadInfo())
+                .isEqualTo(createTransactionReadInfo(
+                        SMALL_BYTES_READ + MEDIUM_BYTES_READ + HIGH_LARGE_BYTES,
+                        3,
+                        ImmutableKvsCallReadInfo.of(KVS_METHOD_NAME_5, HIGH_LARGE_BYTES)));
+
+        assertThat(tracker.getReadInfoByTable())
+                .containsExactlyEntriesOf(ImmutableMap.of(
+                        TABLE_1,
+                        createTransactionReadInfo(
+                                SMALL_BYTES_READ + MEDIUM_BYTES_READ,
+                                2,
+                                ImmutableKvsCallReadInfo.of(KVS_METHOD_NAME_1, SMALL_BYTES_READ))));
+    }
+
+    /**
+     * Schedules tracking calls from {@link #differentMethodCallsAreTracked} repeatedly ({@link #CONCURRENCY_ROUNDS}
```

Comment:
nit: I don't think this comment adds anything :) Is there something you explicitly want to call out for future devs?